crates/ra_project_model/src/lib.rs

```diff
@@ -62,7 +63,12 @@ impl ProjectWorkspace {
             for tgt in pkg.targets(&self.cargo) {
                 let root = tgt.root(&self.cargo);
                 if let Some(file_id) = load(root) {
-                    let crate_id = crate_graph.add_crate_root(file_id);
+                    let edition = if pkg.edition(&self.cargo) == "2015" {
+                        Edition::Edition2015
+                    } else {
+                        Edition::Edition2018
```

Comment:
I've only taken a cursory glance at this but doesn't this default the edition to 2018 when it should be 2015 by default?

Reply:
If I'm reading it correctly, `pkg.edition()` resolves to data that came from `cargo_metadata`. Testing locally, that crate will correctly give `"2015"` edition if none is specified in `Cargo.toml`.  I don't think this is a default, just rather one had to be tested or the other.

Or rather, this defaults to 2018 if a version other than "2015" and "2018" is specified in Cargo.toml, which seems correct. Future editions are more likely to have edition 2018 import semantics than 2015 ones, right?

Reply:
:+1:

Reply:
Exactly. I think this can be made clearer by writing it as a match, though.